app/javascript/components/workflows/professor/exams/accommodations/index.tsx

```diff
@@ -0,0 +1,326 @@
+import { BsPencilSquare } from 'react-icons/bs';
+import { updateAccommodation } from '@hourglass/common/api/professor/accommodations/update';
+import { AlertContext } from '@hourglass/common/alerts';
+import { FaTrash } from 'react-icons/fa';
+import destroyAccommodation from '@hourglass/common/api/professor/accommodations/destroy';
+import Select from 'react-select';
+import { useRegistrationsIndex } from '@hourglass/common/api/professor/registrations';
+import Loading from '@hourglass/common/loading';
+import { createAccommodation } from '@hourglass/common/api/professor/accommodations/create';
+import { DateTime } from 'luxon';
+
+const AccommodationEditor: React.FC<{
+  submit: (startTime: DateTime, extraTime: number) => void;
+  cancel: () => void;
+  accommodation: Accommodation;
+}> = (props) => {
+  const {
+    submit,
+    cancel,
+    accommodation,
+  } = props;
+  const [startTime, setStartTime] = useState(accommodation.startTime);
+  const [extraTime, setExtraTime] = useState(accommodation.extraTime);
+  return (
+    <tr>
+      <td className="align-middle">
+        {accommodation.reg.user.displayName}
+      </td>
+      <td>
+        <DateTimePicker
+          onChange={setStartTime}
+          value={startTime}
+          nullable
+        />
+      </td>
+      <td>
+        <Form.Control
+          type="number"
+          min={0}
+          value={extraTime}
+          onChange={(e) => setExtraTime(Number(e.target.value))}
+        />
+      </td>
+      <td align="right" className="text-nowrap">
+        <Button
+          variant="secondary"
+          onClick={cancel}
+        >
+          Cancel
+        </Button>
+        <Button
+          variant="primary"
+          className="ml-2"
+          onClick={() => submit(startTime, extraTime)}
+        >
+          Save
+        </Button>
+      </td>
+    </tr>
+  );
+};
+
+const SingleAccommodation: React.FC<{
+  refresh: () => void;
+  accommodation: Accommodation;
+}> = (props) => {
+  const {
+    refresh,
+    accommodation,
+  } = props;
+  const [editing, setEditing] = useState(false);
+  const edit = useCallback(() => setEditing(true), []);
+  const stopEdit = useCallback(() => setEditing(false), []);
+  const { alert } = useContext(AlertContext);
+  const destroy = () => {
+    destroyAccommodation(accommodation.id).then((_res) => {
+      refresh();
+      alert({
+        variant: 'success',
+        title: 'Successfully deleted accommodation',
+        autohide: true,
+        message: `Accommodation for '${accommodation.reg.user.displayName}' deleted.`,
+      });
+    }).catch((err) => {
+      alert({
+        variant: 'danger',
+        title: 'Error deleting accommodation',
+        message: err.message,
+      });
+    });
+  };
+  const submit = (startTime: DateTime, extraTime: number) => {
+    updateAccommodation(accommodation.id, {
+      startTime,
+      extraTime,
+    }).then((res) => {
+      if (res.success !== true) {
```

Comment:
You have a check for `res.success !== true` here, but not in `destroyAccommodation` above.  Seems inconsistent.